src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -271,7 +271,16 @@ public Snapshot retrieveSnapshot(UUID snapshotId, boolean onlyRetrieveAvailable)
     @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE, readOnly = true)
     public SnapshotProject retrieveSnapshotProject(UUID snapshotId, boolean onlyRetrieveAvailable) {
         logger.debug("retrieve snapshot id: " + snapshotId);
-        String sql = "SELECT snapshot.id, name, snapshot.profile_id, google_project_id FROM snapshot " +
+        String sql = "SELECT snapshot.id, name, snapshot.profile_id, google_project_id, " +
+            // Select the source dataset project information
+            "(SELECT jsonb_agg(ds)\n" +
+            "  FROM (SELECT d.id, d.name, p.profile_id as \"profileId\", p.google_project_id as \"dataProject\"\n" +
```

Comment:
`google_project_id` is being returned from `project_resource p` which is joined  to the dataset, which is a source of the snapshot.  Right now, it's a single source but eventually it will be multiple sources.

Reply:
Ok, the project id will vary in the future. Why select the other fields if they're not needed?